src/fsharp/infos.fs

```diff
@@ -1526,13 +1526,16 @@ type RecdFieldInfo =
 /// Describes an F# use of a union case
 [<NoComparison; NoEquality>]
 type UnionCaseInfo = 
-    | UnionCaseInfo of TypeInst * Tast.UnionCaseRef 
+    | UnionCaseInfo of TypeInst * Tast.UnionCaseRef * bool
```

Comment:
Could you change the PR to add the flag as Item.UnionCase(ucref,flag) rather than to UnionCaseInfo please? This is because the flag is really a transient thing related to name resolution, whereas the UnionCaseInfo is like a reflection-handle to the union case, representing the logical item.